src/main/JayZ.ts

```diff
@@ -25,69 +31,99 @@ export class JayZ {
         ? config.encryptor
         : new LibsodiumEncryptor()
     this.maxUsesPerDataKey = config.maxUsesPerDataKey || 1
+    this.currentDataKeyUsesRemaining = this.maxUsesPerDataKey
   }
 
-  async encryptItem<T, K extends keyof T>(
-    item: T,
-    fieldsToEncrypt: K[]
-  ): Promise<EncryptedJayZItem<T, K>> {
-    const { dataKey, encryptedDataKey } = await this.getDataKey()
```

Comment:
Note: 

1. In the old code how we had a double `await` here. When we hit an `await` it's a signal to the runtime to "go do other stuff and check back in another event loop tick"

2. Inside of the old `this.getDataKey()` implementation, we'd zero out `this.dataKey` (a pointer) when we switched to a new key. 

So what could happen is: 

1. Caller makes N requests in a function like `itemsToEncrypt.map((item) => { jayz.encryptItem(...) })`

2. We'd go through the loop and trigger N calls to `getDataKey`. Those would resolve. Inside of there we might zero out the pointer to the previous `this.dataKey`

3. So when we hit `this.encrptor.encrypt` on line 35 in a later event loop tick, the pointer to `dataKey` on line 34, might've been zeroed out already.

Reply:
Great explanation!